azure-pipelines.yml

```diff
@@ -1,213 +1,347 @@
+                scriptType: bash
+                scriptLocation: inlineScript
+                inlineScript: |
+                  set -eux  # fail on error
+
+                  az webapp deployment slot create --name $(ANALYZER_DEV_APP_NAME) --resource-group $(DEV_RESOURCE_GROUP_NAME) --configuration-source $(ANALYZER_DEV_APP_NAME) --slot staging
+                  URI=https://$(az webapp config hostname list --webapp-name $(ANALYZER_DEV_APP_NAME) --resource-group $(DEV_RESOURCE_GROUP_NAME) --slot staging --query '[0].name' -o tsv)
+                  echo $URI
+                  echo "##vso[task.setvariable variable=STAGING_URI;isOutput=true]$URI"
+            - task: AzureWebAppContainer@1
+              displayName: Deploy to staging slot
+              inputs:
+                azureSubscription: $(DEV_AZURE_SUBSCRIPTION)
+                deployToSlotOrASE: true
+                resourceGroupName: $(DEV_RESOURCE_GROUP_NAME)
+                appName: $(ANALYZER_DEV_APP_NAME)
+                slotName: staging
+                imageName: $(REGISTRY_NAME)presidio-analyzer:$(Build.BuildId)
+
+        - job: DeployAnonymizerToDev
+          displayName: Deploy Anonymizer to Dev Environment
+          pool:
+            vmImage: 'ubuntu-16.04'
+          steps:
+            - checkout: none
+            - task: AzureCLI@2
+              displayName: Create staging slot
+              name: createStagingSlot
+              inputs:
+                azureSubscription: $(DEV_AZURE_SUBSCRIPTION)
+                scriptType: bash
+                scriptLocation: inlineScript
+                inlineScript: |
+                  set -eux  # fail on error
+
+                  az webapp deployment slot create --name $(ANONYMIZER_DEV_APP_NAME) --resource-group $(DEV_RESOURCE_GROUP_NAME) --configuration-source $(ANONYMIZER_DEV_APP_NAME) --slot staging
+                  URI=https://$(az webapp config hostname list --webapp-name $(ANONYMIZER_DEV_APP_NAME) --resource-group $(DEV_RESOURCE_GROUP_NAME) --slot staging --query '[0].name' -o tsv)
+                  echo $URI
+                  echo "##vso[task.setvariable variable=STAGING_URI;isOutput=true]$URI"
+            - task: AzureWebAppContainer@1
+              displayName: Deploy to staging slot
+              inputs:
+                azureSubscription: $(DEV_AZURE_SUBSCRIPTION)
+                deployToSlotOrASE: true
+                resourceGroupName: $(DEV_RESOURCE_GROUP_NAME)
+                appName: $(ANONYMIZER_DEV_APP_NAME)
+                slotName: staging
+                imageName: $(REGISTRY_NAME)presidio-anonymizer:$(Build.BuildId)
+        
+        - job: WaitForDeployment
+          displayName: Wait for deployment
+          dependsOn: 
+            - DeployAnalyzerToDev
+            - DeployAnonymizerToDev
+          pool: Server
+          steps:
+          - task: Delay@1
+            inputs:
+              delayForMinutes: '5'
```

Comment:
Isn't there any other way?
What if it won't finish in 5 minutes?

Reply:
i wish there is, and would take any suggestion.
the problem is that its an app service pulling and running the container. there is no "status" or "readiness" you can query.
the only workaround which may fit what you want is to put the smoke tests in a a while loop. 
do you think that's better?

Reply:
how about adding "semi" smoke test in a loop - curl to a health check to each service?
question - can we deploy the apps using ARM templates and wait for an output from the template (say have an output variable with the slot id)?

Reply:
@navalev the ARM template deployment exists after the app service is up and does not wait for the container to pull. there is also no "readiness" that i could find that indicates if the container is ready or not. i will do the following:
* implement a while loop on the smoke test instead of delay, with an exit of 5 minutes.
* ask a question in the relevant CSE tech domain

Reply:
so we're going with the while loop (until we get a 200)

Reply:
done